src/borg/testsuite/archiver.py

```diff
@@ -3630,6 +3630,15 @@ def test_change_location_does_not_change_algorithm_argon2(self):
     def test_change_location_does_not_change_algorithm_pbkdf2(self):
         self.verify_change_location_does_not_change_algorithm('pbkdf2', b'sha256')
 
+    def test_key_change_algorithm(self):
+        self.cmd('init', '--encryption=repokey', '--key-algorithm=pbkdf2', self.repository_location)
+
+        self.cmd('key', 'change-algorithm', self.repository_location, 'argon2')
+
+        with Repository(self.repository_path) as repository:
+            _, key = Manifest.load(repository, Manifest.NO_OPERATION_CHECK)
+        assert key._encrypted_key_algorithm == 'argon2 aes256-ctr hmac-sha256'
+
```

Comment:
maybe you could do a `borg info` here to actually use the new key / check whether it works.

Reply:
Good idea! Done.